UI for follow-ups in Interactions by Service report

leahjlou opened the pull request:
I noticed a couple of weird things while adding this UI.

1. Follow-ups aren't required to have an associated service. The follow-ups with unspecified services don't show up anywhere in this report or in the total. It seems misleading/confusing to me. Thoughts on adding an "unspecified" row? (this would also require a backend change, happy to do that)

2. If a follow-up is associated with multiple services, the "grand total" counts that same follow-up twice in the sums for count & duration. It seems like we want to change the backend reporting logic here -- I'm happy to do this, but because it's a nontrivial change I wanted to check if there is any reason we might want to keep this or maybe even change a follow-up to only be associated with 1 service.

For example:

Created a follow up for a client with two different services selected:
<img width="506" alt="Screen Shot 2020-12-22 at 6 25 17 PM" src="https://user-images.githubusercontent.com/2423414/102948777-ecd94a80-4483-11eb-8980-9ca1ec810879.png">

Report shows follow up count/duration correctly on individual line items, but the total counts the same follow-up twice:
<img width="1122" alt="Screen Shot 2020-12-22 at 6 25 54 PM" src="https://user-images.githubusercontent.com/2423414/102948780-ee0a7780-4483-11eb-9640-d7695f1aea68.png">

joeldenning commented:
> Thoughts on adding an "unspecified" row?

Yes let's do that.

> The "grand total" counts that same follow-up twice

Yes this is a problem that we should fix. Associating a follow up with multiple services is as-designed. Not sure the best way to do this. Maybe just the grand total row is corrected?

leahjlou commented:
@joeldenning 

> Associating a follow up with multiple services is as-designed. Not sure the best way to do this. Maybe just the grand total row is corrected?

Yeah, I think just correcting the grand totals to not double-count the followups would be good. I'll add that in on this branch (and the unspecified row). Thanks for your input!

leahjlou commented:
Added an unspecified row for follow-ups that aren't associated with a program/service, and corrected the total row to display the real total (not double-counting a single follow-up).

It is working correctly -- the end experience feels a little confusing to me, but it's probably OK. In these screenshots, you can see that the totals row is correctly summing the follow-ups, but it might seem weird that it doesn't actually display the total if you sum the columns above, because in this case, one of the follow-ups is associated with two different services, and two of the follow-ups are unspecified. I think it's okay, but wanted to run it by you. I can't think of a way to display the data that is more accurate than this.

<img width="1466" alt="Screen Shot 2020-12-29 at 12 10 00 PM" src="https://user-images.githubusercontent.com/2423414/103308189-10921880-49cf-11eb-8ef3-411823c932a2.png">
<img width="1087" alt="Screen Shot 2020-12-29 at 12 11 43 PM" src="https://user-images.githubusercontent.com/2423414/103308200-138d0900-49cf-11eb-87d2-99918410300e.png">

joeldenning reviewed (approved):
Thanks
